File: trunk/tests/CoreTests/OutRefParameters/Class4.cs
using SharpKit.JavaScript;

namespace SharpKitWebApp5
{
    [JsType(JsMode.Prototype, NativeOverloads = false, NativeConstructors = true)]
    public class Foo2
    {
        public void test()
        {
            JsNumber b;
            foo1(1, out b);
        }

        public void foo1(JsNumber a, out JsNumber outB)
        {
            foo2(1, out outB);
            outB = 7;
        }

        public void foo2(JsNumber a, out JsNumber outC)
        {
            outC = 2;
        }
    }
}

File: tests/CoreTests/OutRefParameters/Class4.js

if (typeof(SharpKitWebApp5) == "undefined")
    var SharpKitWebApp5 = {};
SharpKitWebApp5.Foo2 = function (){
};
SharpKitWebApp5.Foo2.prototype.test = function (){
    var b;
    (function (){
        var $1 = {
            Value: b
        };
        var $res = this.foo1(1, $1);
        b = $1.Value;
        return $res;
    }).call(this);
};
SharpKitWebApp5.Foo2.prototype.foo1 = function (a, outB){
    this.foo2(1, outB);
    outB.Value = 7;
};
SharpKitWebApp5.Foo2.prototype.foo2 = function (a, outC){
    outC.Value = 2;
};

